tests/bluetooth/tester/src/audio/btp_pbp.c

```diff
@@ -0,0 +1,267 @@
+{
+	uint8_t *broadcast_name_len = user_data;
+
+	switch (data->type) {
+	case BT_DATA_BROADCAST_NAME:
+		*broadcast_name_len = data->data_len;
+		return false;
+	default:
+		return true;
+	}
+}
+
+static bool scan_get_data(struct bt_data *data, void *user_data)
+{
+	enum bt_pbp_announcement_feature source_features;
+	uint32_t broadcast_id;
+	uint8_t metadata_len;
+	uint8_t *metadata;
+	struct btp_pbp_ev_public_broadcast_anouncement_found_rp *ev = user_data;
+
+	switch (data->type) {
+	case BT_DATA_BROADCAST_NAME:
+	ev->broadcast_name_len = data->data_len;
+		memcpy(ev->broadcast_name, data->data, data->data_len);
+		return false;
+	case BT_DATA_SVC_DATA16:
+		broadcast_id = sys_get_le24(data->data + BT_UUID_SIZE_16);
+		sys_put_le24(broadcast_id, ev->broadcast_id);
+		metadata_len = bt_pbp_parse_announcement(data, &source_features, &metadata);
+		ev->pba_features = source_features;
+		return false;
+	default:
+		return true;
+	}
+}
+
+static void pbp_scan_recv(const struct bt_le_scan_recv_info *info, struct net_buf_simple *ad)
+{
+	if ((info->adv_props & BT_GAP_ADV_PROP_CONNECTABLE) || info->interval == 0) {
+		return;
+	}
+
+	uint8_t broadcast_name_len = 0;
+	struct net_buf_simple ad_copy;
+
+	net_buf_simple_clone(ad, &ad_copy);
+	bt_data_parse(&ad_copy, scan_get_broadcast_name_len, &broadcast_name_len);
+
+	struct btp_pbp_ev_public_broadcast_anouncement_found_rp *ev_ptr;
+
+	tester_rsp_buffer_lock();
+	tester_rsp_buffer_allocate(sizeof(*ev_ptr) + broadcast_name_len, (uint8_t **)&ev_ptr);
+
+	bt_addr_le_copy(&ev_ptr->address, info->addr);
+	ev_ptr->advertiser_sid = info->sid;
+	ev_ptr->padv_interval = info->interval;
+	bt_data_parse(ad, scan_get_data, ev_ptr);
+
+	tester_event(BTP_SERVICE_ID_PBP, BTP_PBP_EV_PUBLIC_BROADCAST_ANOUNCEMENT_FOUND, ev_ptr,
+		     sizeof(*ev_ptr) + broadcast_name_len);
+
+	tester_rsp_buffer_free();
+	tester_rsp_buffer_unlock();
+}
+
+static struct bt_le_scan_cb pbp_scan_cb = {
+	.recv = pbp_scan_recv,
+};
+
+static uint8_t pbp_read_supported_commands(const void *cmd, uint16_t cmd_len, void *rsp,
+					   uint16_t *rsp_len)
+{
+	struct btp_pbp_read_supported_commands_rp *rp = rsp;
+
+	tester_set_bit(rp->data, BTP_PBP_READ_SUPPORTED_COMMANDS);
+	tester_set_bit(rp->data, BTP_PBP_SET_PUBLIC_BROADCAST_ANNOUNCEMENT);
+	tester_set_bit(rp->data, BTP_PBP_SET_BROADCAST_NAME);
+	tester_set_bit(rp->data, BTP_PBP_BROADCAST_SCAN_START);
+	tester_set_bit(rp->data, BTP_PBP_BROADCAST_SCAN_STOP);
+
+	*rsp_len = sizeof(*rp) + 1;
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static int pbp_broadcast_source_adv_setup(void)
+{
+	struct bt_le_adv_param param =
+		BT_LE_ADV_PARAM_INIT(0, BT_GAP_ADV_FAST_INT_MIN_2, BT_GAP_ADV_FAST_INT_MAX_2, NULL);
+	uint32_t gap_settings = BIT(BTP_GAP_SETTINGS_DISCOVERABLE) |
+				BIT(BTP_GAP_SETTINGS_EXTENDED_ADVERTISING);
+	uint32_t broadcast_id;
+
+	NET_BUF_SIMPLE_DEFINE(ad_buf, BT_UUID_SIZE_16 + BT_AUDIO_BROADCAST_ID_SIZE);
+	NET_BUF_SIMPLE_DEFINE(pba_buf, BT_UUID_SIZE_16 + 3 + pbp_metadata_cached_len);
```

Comment:
+ 3 is somewhat magic. Can it be defined?

Reply:
You are right, it should have been 2, and I added a definition